ui-tests/cypress/integration/components/ux_fragments.spec.js

```
@@ -1,141 +1,783 @@
+    describe('Fragments Browsing', () => {
+
+      it([Tag.FEATURE, 'ENG-3522'], 'Search a fragment from his fragment code', () => {
+        addTestFragment();
+        openFragmentsPage()
+            .then(page =>
+                page.getContent().getSearchCodeInput().then(input => page.getContent().type(input, fragment.code.substring(3, 7))))
+            .then(page => page.getContent().clickSearchSubmitButton())
+            .then(page =>
+                page.getContent().getTableRows().should('have.length', 1)
+                    .each(row => cy.get(row).children(htmlElements.td).eq(0).should('contain', fragment.code.substring(3, 7))));
+        cy.get('@currentPage')
+          .then(page => {
+            page.getContent().getPagination().getItemsCurrent().invoke('text').should('be.equal', '1-1');
+            page.getContent().getPagination().getItemsTotal().invoke('text').should('be.equal', '1');
+          });
+      });
 
-      cy.validateToast(currentPage);
+      it([Tag.FEATURE, 'ENG-3522'], 'Bug: Widget type does not filter', () => {
+        openFragmentsPage()
+            .then(page => {
+              page.getContent().getWidgetFilter().select('Content');
+              page.getContent().clickSearchSubmitButton();
+            })
+            .then(page =>
+                page.getContent().getTableRows().should('have.length', 1)
+                    .each(row => cy.get(row).children(htmlElements.td).eq(2).should('have.text', 'Content')));
+        cy.get('@currentPage')
+          .then(page => {
+            page.getContent().getPagination().getItemsCurrent().invoke('text').should('be.equal', '1-1');
+            page.getContent().getPagination().getItemsTotal().invoke('text').should('be.equal', '1');
+          });
+      });
 
-      cy.wrap(null).as('fragmentToBeDeleted');
-    });
+      //TODO remove x when bug is fixed
+      xit([Tag.FEATURE, 'ENG-3522'], 'Search for a fragment through the widget filter', () => {
```

what's the bug? is it a known issue? is it a test bug? why do we want to hide the failure of a test?

---

The failure of this test is randomic: it depends on the type of widget chosen. 
Most of the time it passes because it doesn't catch the widget where the malfunctioning is fully apparent. 
This is the reason why I wrote a specific test to detect the imperfect filtering.
 I decided to mute the test until it is fixed only for this reason.